Calculator testing completion

RNtaate opened the pull request:
## What does this PR do?
- [X] Installs react-testing library, jest-dom, react-test-renderer testing dependencies.
- [X] Creates a __test__ folder in the components folder to hold all test files for the components.
- [X] Creates tests for various components using jest and React-testing-library such as App, Button, Home, Quote, ButtonPanel, e.t.c
- [X] Performs snapshot testing for various components with jest.
- [X] Creates a __test__ folder in the logics folder and adds the calculate test file to it.
- [X] Adds unit tests for the calculate.js file.
- [X] Fixes eslint linter offenses.

Torres-ssf commented on src/logic/__test__/calculate.test.js:
```diff
@@ -0,0 +1,45 @@
+  expect(typeof newData).toBe('object');
+});
+
+test('returns an object with calculated property values basing on buttonName provided as argument', () => {
+  let newData = calculate('3', myData);
+  expect(newData.next).toBe('3');
+  expect(newData.next).not.toBeNull();
+
+  myData.next = '3';
+  newData = calculate('+', myData);
+  expect(newData.total).toBe('3');
+  expect(newData.total).not.toBeNull();
+  expect(newData.next).toBeNull();
+  expect(newData.operation).toBe('+');
+
+  myData.total = '3';
+  myData.operation = '+';
+  myData.next = '2';
+  newData = calculate('=', myData);
+  expect(newData.total).toBe('5');
+  expect(newData.next).toBeNull();
+  expect(newData.operation).toBeNull();
+});
```
- [ ] We need to add a unit test for each math operator `( +, -, *, /, %)`. Example: the test should ensure that `5 * 10 = 50`.

- [ ] We also need to add a unit test to ensure that the buttons like `(+/-, AC, =, ., )` are working as expected. Example: `+/-` should turn `-4` into `4`

Torres-ssf reviewed (changes requested):
### Changes Required :muscle: 
Hi @RNtaate :hand: 
Great job testing your application :clap: 
I can see that a lot of tests were implemented. That is really nice.
But we need to more unit tests to our calculate.js. The reason for this is to make sure that all operations from calculate.js are working properly.
![Screenshot from 2021-04-12 11-26-20](https://user-images.githubusercontent.com/30977845/114410702-f5b07200-9b81-11eb-952a-528bcef47f46.png)


- [ ] Check my comment below

Happy coding :computer: 
**Sérgio Torres**

RNtaate commented:
## Changes made
- [X] Added unit test for the various math operators as requested by the reviewer.
- [X] Added unit tests for buttons like `AC`, `+/-`, `.`

Torres-ssf reviewed (approved):
### PR Approved :+1: :clap: 
Hello again, @RNtaate :hand: 
Nice job adding more unit tests 
You may merge the PR :+1: 
Good luck with the next project :smile: 

Happy coding :computer: 
**Sérgio Torres**
